Supertext.Base/Factory/AutofacKeyFactory.cs

```diff
@@ -59,16 +59,49 @@ public T CreateComponent(TKey key)
             throw new KeyNotFoundException($"No component registered with key '{key}'");
         }
 
+        public T CreateComponent(TKey key, params Parameter[] parameters)
```

Comment:
What about providing generic method overloads in order to not expose Autofac parameters?
Like:
* `public T CreateComponent<T1>(TKey key, T1 parameter1)`
* `public T CreateComponent<T1, T2>(TKey key, T1 parameter1, T2 parameter2)`
* etc until 5 or so...

Reply:
I thought about this too but then we limit it to TypedParameter. So for NamedParameter or ResolvedParameter we would need additional methods like CreateComponentWithNamedParameters. Or de we just leave those type of parameters out?

Reply:
> I thought about this too but then we limit it to TypedParameter.

True. In most cases `TypedParameter` is enough. What about the generic way (as suggested) and a custom abstraction for `TypedParameter` and `NamedParameter`?